src/services/item/plugins/action/service.ts

```diff
@@ -54,11 +56,18 @@ export class ActionItemService {
     filters: { view?: Context; sampleSize?: number },
   ): Promise<Action[]> {
     const { view = Context.Builder, sampleSize = DEFAULT_ACTIONS_SAMPLE_SIZE } = filters;
+
     // get item
     const item = await this.itemService.get(actor, repositories, itemId);
```

Comment:
Information: This line automatically check the actor has access to the item. It means that it will return if the actor is undefined and the item is public.